gpcontrib/gp_orca/gporca.c

```diff
@@ -1,16 +1,117 @@
 #include "postgres.h"
 
 #include "cdb/cdbvars.h"
+#include "executor/executor.h"
 #include "miscadmin.h"
+#include "optimizer/orca.h"
+#include "optimizer/planner.h"
+#include "storage/ipc.h"
 #include "utils/builtins.h"
 #include "utils/guc.h"
 #include "utils/vmem_tracker.h"
+#include "utils/memutils.h"
 
 PG_MODULE_MAGIC;
 
+extern void InitGPOPT();
+extern void TerminateGPOPT();
+
+extern void compute_jit_flags(PlannedStmt* pstmt);
+
 void		_PG_init(void);
 void		_PG_fini(void);
 
+static planner_hook_type prev_planner = NULL;
+
+static void
+gp_orca_shutdown(int code, Datum arg)
+{
+	TerminateGPOPT();
+
+	if (OptimizerMemoryContext != NULL)
```

Comment:
To keep up with the style:
```suggestion
	if (NULL != OptimizerMemoryContext)
```